source/slang-capture-replay/capture_utility.cpp

```diff
@@ -0,0 +1,66 @@
+
+#include <cstdlib>
+#include <cstring>
+#include <string>
+#include <stdarg.h>
+#include <mutex>
+
+#include "capture_utility.h"
+
+constexpr const char* kCaptureLayerEnvVar = "SLANG_CAPTURE_LAYER";
+constexpr const char* kCaptureLayerLogLevel = "SLANG_CAPTURE_LOG_LEVEL";
+
+namespace SlangCapture
+{
+    static unsigned int g_logLevel = LogLevel::Silent;
+
+    bool isCaptureLayerEnabled()
+    {
+        if (const char* envVar = std::getenv(kCaptureLayerEnvVar))
+        {
+            if (std::strcmp(envVar, "1") == 0)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    void setLogLevel()
+    {
+        static std::mutex logLevelMutex;
+        {
+            std::lock_guard<std::mutex> scopeLock(logLevelMutex);
+
+            // We only want to set the log level once
+            if (g_logLevel != LogLevel::Silent)
+            {
+                return;
+            }
+
+            if (const char* envVar = std::getenv(kCaptureLayerLogLevel))
+            {
+                char* end = nullptr;
+                unsigned int logLevel = std::strtol(envVar, &end, 10);
+                if (end && (*end == 0))
+                {
+                    g_logLevel = std::min((unsigned int)(LogLevel::Verbose), logLevel);
+                    return;
+                }
+            }
+        }
+    }
+
+    void slangCaptureLog(LogLevel logLevel, const char* fmt, ...)
+    {
+        if (logLevel > g_logLevel)
+        {
+            return;
+        }
+
+        va_list args;
+        va_start(args, fmt);
+        vfprintf(stdout, fmt, args);
+        va_end(args);
```

Comment:
Are we going to generate multiple files each of which will work for one thread?
Or are we going to generate one big log file that contains all?

Reply:
I'm going to generate one log file per thread, so we don't need to handle any thread issues.